### PR TITLE
NAC-1768: Added handling of 500 Errors in APIM

### PR DESCRIPTION
**Pull request by ipiatrou**
## What’s changing?

All details of 500 response have been removed. Created a generic error with no information.
image

## Why?

NAC-1768

## This PR has:

- [x] **JIRA ticket(s)** (_Must be listed in PR request and branch name should follow convention NAC-123-description_)<description>

- [x] **Security Related Changes** (_Security Tab questions have been answered on JIRA ticket(s)_)

- [ ] **Commented source code** (_required on public classes and methods, and elsewhere as appropriate_)

- [ ] **Documented API changes** (_required on public classes and methods, and elsewhere as appropriate_)
    - [ ] External API documentation - OpenApi docs and Generated Docs (e.g. https://github.com/18F/piipan/tree/dev/docs/openapi) 
    - [ ] Internal API documentation (e.g. https://github.com/18F/piipan/tree/dev/metrics/docs/openapi)   

- [ ] **Developer documentation** (_for new build/test/API changes or complex portions of the system_)

- [ ] **Automated unit tests** (_to maintain or increase level of code coverage_)

- [x] **Changes to IAC** (_add any deployment steps that require manual intervention to the draft release notes_)
    - [x] Have IaC scripts run successfully against an existing deployed environment?
    - [ ] Have IaC scripts run successfully against a newly deployed environment?
    - [ ] Have any new bash scripts been given the appropriate permissions? (e.g. git update-index --chmod=+x $script)
    - [ ] Do the IaC changes require deleting existing resources?

## Anything else?

_Add other details that are helpful for review here_

## Code Review Checklist:

- [ ] Automated checks all pass? (code builds successfully, tests pass, test coverage meets/exceeds threshold, etc.) 
- [ ] Have all requirements and Acceptance Criteria been met?
- **If there are code changes**
    - [ ] Is there no unnecessary complexity? Code is readable, maintainable, etc.
    - [ ] Do the changes follow SOLID principles?
    - [ ] Are the changes robust? Concurrency considered? Performance? Security (e.g. no sql injections)?
    - [ ] Are newly added dependencies needed and appropriate? Are dependency files (e.g. packages.json & packages.lock.json) updated?
    - [ ] For any new projects/solutions, have they been added to code coverage files?
    - [ ] Public classes & methods are documented    
- **If there are API changes**
    - [ ] Does API make sense? Useful and concise?
    - [ ] Are existing API interfaces/contracts intact? i.e. no breaking changes
    - [ ] Are API tests included?
    - [ ] Are API responses appropriate and documented? (200s, 201, 404s, etc.)
    - [ ] Are internals kept private and not leaked into the API?
    - [ ] External API documentation - OpenApi docs and Generated Docs 
    - [ ] Internal API documentation    
- **If there are IAC changes**
    - [ ] IAC Scripts are readable & maintainable
    - [ ] IAC Scripts are documented
    - [ ] IAC scripts pass required checks
    - [ ] IAC scripts were run successfully by author
- **General Logging Questions**
    - [ ] Do new capabilities (or changed capabilities) include logging?
    - [ ] Is PII not logged?
- **General Error Handling Questions**
    - [ ] Has exception handling been added for new capabilities (or changed capabilities)?
    - [ ] Are errors/exceptions properly communicated to end-users? Were there requirements for user facing error messages?
    - [ ] Are errors/exceptions logged?
    - [ ] Is PII kept out of error/exception logging?
- **If there are Developer Documentation Changes**
    - [ ] General Project documentation (under piipan\docs. e.g. iac.md, architecture.md, etc.)
    - [ ] Github documentation additions/changes (Readmes)
- **General automated test questions**
    - [ ] End-to-End Tests provided (if code implements new user story)
    - [ ] Are tests sufficient?
    - [ ] Corner cases considered?
    - [ ] Are performance tests needed?

### CODE REVIEW

**Comment by dborgen2** [comment]
@ipiatrou Can you add @ryangeorgeharvey as a reviewer here? He usually has more insights for IAC changes. It looks good to me, but I'd be curious what he says before it gets approved. I assume you tested these changes with multiple APIs, not just find_matches?

File: iac/apim-policy.xml [hunk 1]
```diff
@@ -30,5 +30,15 @@
         <set-header name="Strict-Transport-Security" exists-action="override">    
             <value>Strict-Transport-Security: max-age=31536000; includeSubDomains; preload</value>
         </set-header>  
+        <choose>
+            <when condition="@(context.Response.StatusCode >= 500)">
+                <set-body>@{
+                    return new JObject(
+                            new JProperty("statusCode",  context.Response.StatusCode),
+                            new JProperty("message", "Something went wrong.")
```
Comment (ryangeorgeharvey) [inline iac/apim-policy.xml:38]: Suggestion (non-blocking): I think we should be using a better message then "Something went wrong". This is the message that states will receive when getting a 500 error. I would check with Bill for guidance, but maybe something like "The system has encountered an error. If the error persists, please contact NAC Helpdesk."

Comment (ryangeorgeharvey) [reply iac/apim-policy.xml:38]: @bgfoshay ^^

Comment (bgfoshay) [reply iac/apim-policy.xml:38]: Let's go with "The system has encountered an error. Please try your request again. If the error persists, please contact the NAC Helpdesk."

**Comment by ipiatrou** [comment]
> @ipiatrou Can you add @ryangeorgeharvey as a reviewer here? He usually has more insights for IAC changes. It looks good to me, but I'd be curious what he says before it gets approved. I assume you tested these changes with multiple APIs, not just find_matches?

File: iac/apim-policy.xml (hunk 1 above)
Comment (ipiatrou) [reply iac/apim-policy.xml:38]: @ryangeorgeharvey Agree, the message can be better.

**Comment by ipiatrou** [comment]
> @ipiatrou Can you add @ryangeorgeharvey as a reviewer here? He usually has more insights for IAC changes. It looks good to me, but I'd be curious what he says before it gets approved. I assume you tested these changes with multiple APIs, not just find_matches?

@dborgen2 Sure, I tested these changes with multiple APIs.

File: iac/apim-policy.xml (hunk 1 above)
Comment (ipiatrou) [reply iac/apim-policy.xml]: @ryangeorgeharvey @bgfoshay 
Fixed.
image
image

**Review by ryangeorgeharvey — APPROVED** [review]
LGTM
